rmf_demos_tasks/README.md

```diff
@@ -2,6 +2,92 @@
+
+  Example for Office world:
+  ```
+  ros2 run rmf_demos_tasks dispatch_delivery -p pantry -pd coke_dispenser -d hardware_2 -di coke_ingestor --use_sim_time
+  ```
+- `dispatch_loop`
+
+  This script submits a loop task. It takes in a start `-s` and finish `-f` waypoint, as well as an optional loop number `-n` that defaults to 1.
+
+  Example for Clinic world:
+  ```
+  ros2 run rmf_demos_tasks dispatch_loop -s L1_right_nurse_center -f L1_left_nurse_center -n 2 --use_sim_time
+  ```
+
+Do remember to add the `--use_sim_time` flag when running these tasks in simulation.
+
+## Flexible Tasks
+
+The new task system allows users to construct and submit their own tasks in a more flexible manner, such as having multiple stops in a loop or specifying a robot for a cleaning task. This package contains some helpful scripts that demonstrate how to compose such tasks:
+- `direct_patrol`
+
+  This task is similar to `dispatch_loop`, but it allows users to specify a robot to perform the patrol task. On top of the start and finish waypoints, you will need to provide a fleet name `-F` and robot name `-R`.
+
+  Example for Office world:
+  ```
+  ros2 run rmf_demos_tasks direct_patrol -F tinyRobot -R tinyRobot1 -s pantry -f lounge -n 2 --use_sim_time
+  ```
+- `dispatch_clean`
+
+  This script submits a cleaning task. The `-cs` flag takes in the desired cleaning zone as an argument. You may also choose to specify a robot to perform the cleaning task by providing a fleet name `-F` and robot name `-R`.
+
+  Example for Hotel world:
+  ```
+  ros2 run rmf_demos_tasks dispatch_clean -cs clean_lobby -F cleanerBotA -R cleanerBotA_1 --use_sim_time
+  ```
+- `dispatch_multi_delivery`
+
+  This script allows users to perform multiple deliveries at one go. It takes in information about the pickup `-p` and dropoff `-d` activities in the order you submit them. For each activity, you will need to provide the place, handler, payload SKU and quantity separated by a comma.
+
+  Example for Office world:
+  ```
+  ros2 run rmf_demos_tasks dispatch_multi_delivery -p pantry,coke_dispenser,coke,1 -d hardware_2,coke_ingestor,coke,1 --use_sim_time
+- `dispatch_multi_stop`
+
+  This script takes in multiple places `-p` for the robot to travel to.
+
+  Example for Office world:
+  ```
+  ros2 run rmf_demos_tasks dispatch_multi_stop -p supplies pantry coe --use_sim_time
+  ```
+- `dispatch_teleop`
+
+  This script demonstrates how to compose a task with the `PerformAction` feature. You can ask a robot to go to a starting point `-s` to execute an action.
+
+  Example for Office world:
+  ```
+  ros2 run rmf_demos_tasks dispatch_teleop -F tinyRobot -R tinyRobot1 -s coe --use_sim_time
+  ```
+
+  After completing your teleop action, you will need to publish the following message to return control to RMF:
+  ```
+  ros2 topic pub /action_execution_notice rmf_fleet_msgs/msg/ModeRequest '{fleet_name:  tinyRobot, robot_name: tinyRobot1, mode: {mode: 0}}' --once
+  ```
+
```

Comment:
Should we also include the usage of `dispatch_multi_action`?

Reply:
Got it, updated!